appdev/guides/pushnotifications.rst

```diff
@@ -0,0 +1,235 @@
+Push notifications
+==================
+
+Let's assume that you have an app created with Clickable and published on the OpenStore. But now you want to be able to send Push Notifications to your users. First of all, you need to understand how this is working:
+
+.. image:: ../../_static/images/appdev/guides/pushnotifications/Diagram.png
+
+1. Every app which uses push notifications has got a unique token. This token identifies the user, the device and the app itself. The token is generated by the UBports Push Service.
+2. You will need the token to send a push notification so the app needs to send this token to the entity, which will send the notification.
+3. With the token you can send a simple HTTP request to the UBports Push Server which will forward the notification the user's device.
+
+Let's practice this step-by-step.
+
+.. note::
+    In the following example we will not implement a server. Also the communication between your app and your server is up to you. Please inform the user about the communication with your server by providing a privacy policy!
+
+Make the app ready for push notifications
+-----------------------------------------
+
+Implementing the PushClient
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First we need to add the policy group „push-notification-client“. Your apparmor file could look like this::
+
+	{
+	    "policy_groups": [
+		"networking",
+		"push-notification-client"
```

Comment:
In code examples, it may be helpful to highlight certain lines. That could be useful here to show which line was added.

To do this, use the `code-block` directive rather than `::`. Then, add the `:emphasize-lines:` argument to the directive like this:

```
.. code-block:: js
   :emphasize-lines: 3

    ... some code ...
```

http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

Reply:
I've added some code highlighting and highlighted lines now

Reply:
Oops, sorry. Is fixed